propagation-stackdriver/pom.xml

```diff
@@ -25,13 +25,63 @@
   <artifactId>brave-propagation-stackdriver</artifactId>
 
   <properties>
+    <module.name>brave.propagation.stackdriver</module.name>
+
     <main.basedir>${project.basedir}/..</main.basedir>
   </properties>
 
   <dependencies>
     <dependency>
-      <groupId>io.zipkin.brave</groupId>
+      <groupId>${brave.groupId}</groupId>
       <artifactId>brave</artifactId>
     </dependency>
   </dependencies>
+
+  <build>
+    <plugins>
+      <!-- Shade internal types -->
+      <plugin>
+        <artifactId>maven-shade-plugin</artifactId>
+        <version>${maven-shade-plugin.version}</version>
+        <executions>
+          <execution>
+            <phase>package</phase>
+            <goals>
+              <goal>shade</goal>
+            </goals>
+            <configuration>
+              <createDependencyReducedPom>false</createDependencyReducedPom>
+              <artifactSet>
+                <includes>
+                  <include>${brave.groupId}:brave</include>
+                </includes>
+              </artifactSet>
+              <filters>
+                <filter>
+                  <artifact>${brave.groupId}:brave</artifact>
+                  <includes>
+                    <include>brave/internal/propagation/StringPropagationAdapter*.class</include>
+                    <include>brave/internal/codec/HexCodec*.class</include>
```

Comment:
@elefeint no more bomb :D

Reply:
Thank you! This is great; I did not consider shading as a solution; I kept trying to figure out how to enforce version uniformity in spring-cloud-gcp.

Reply:
some internal things aren't written to be shaded well, these were.. sometimes you can lose time in the shade :D

Reply:
ex anything that references a thread local can't be shaded mostly